### PR TITLE
PDKEnclosures

### PR DESCRIPTION
**Pull request by sebastian-goeldi**
Allows to use a colllection of Enclosures to be applied to a cell

### CODE REVIEW

**Comment by sebastian-goeldi** [comment]
@joamatab any ideas for names on the enclosure set naming? I am not sure what to name it. It allows you to define a set of enclosures and they can be applied to a cell and will automatically create all the enclosures defined in it.

**Comment by joamatab** [comment]
i like the name  `Pdk.enclosures` 

`special_enclosures` is a bit wordy :)

**Comment by sebastian-goeldi** [comment]
I think I didn't explain properly:

So, with the new one, you have one default one that depending on what "main_layers" of each enclosure are present, will apply any enclosure in it. Special enclosures would be if you need a non-standard one, so I think it would be fine to have a longer name. I mainly meant for the new class (currently named `FullEnclosure`).

The standard Enclosure collection would be in `Pdk.enclosure`

**Comment by sebastian-goeldi** [comment]
* renamed `Enclosure` -> `LayerEnclosure`
* renamed `PdkEnclosure` -> `CellEnclousre`
* renamed `Pdk.enclosures` -> `Pdk.layer_enclosures`
* added `Pdk.enclosure` which is the CellEnclosure to use for the Pdk

TODO:

`PortHoles` (carved out in enclosures), are not selective for enclosures from their ports, yet.
